homeassistant/components/whois/sensor.py

```diff
@@ -39,33 +41,52 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the WHOIS sensor."""
-    domain = config[CONF_DOMAIN]
-    name = config[CONF_NAME]
-
+    LOGGER.warning(
+        "Configuration of the Whois platform in YAML is deprecated and will be "
+        "removed in Home Assistant 2022.4; Your existing configuration "
+        "has been imported into the UI automatically and can be safely removed "
+        "from your configuration.yaml file"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={CONF_NAME: config[CONF_NAME]},
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the platform from config_entry."""
+    domain = entry.data[CONF_DOMAIN]
     try:
-        if "expiration_date" in whois.whois(domain):
-            add_entities([WhoisSensor(name, domain)], True)
-        else:
-            LOGGER.error(
-                "WHOIS lookup for %s didn't contain an expiration date", domain
-            )
-            return
+        info = await hass.async_add_executor_job(whois.whois, domain)
     except whois.BaseException as ex:  # pylint: disable=broad-except
```

Comment:
Not sure why `BaseException` is being used here, this should at least be `Exception` or better `PywhoisError`. But you probably already knew that. Replacing the library will of course fix it too.

Reply:
That is existing code not related to this pr